addon/authenticators/oauth2-password-grant.js

```diff
@@ -279,35 +279,34 @@ export default BaseAuthenticator.extend({
     return new RSVP.Promise((resolve, reject) => {
       const data = { 'grant_type': 'password', username: identification, password };
       const serverTokenEndpoint = this.get('serverTokenEndpoint');
-      const useResponse = this.get('rejectWithResponse');
 
-      if (!useResponse) {
-        deprecate('Ember Simple Auth: The default value of false for the rejectWithResponse property should no longer be relied on; instead set the property to true to enable the future behavior.', false, {
-          id: `ember-simple-auth.authenticator.no-reject-with-response`,
-          until: '3.0.0'
-        });
-      }
+      const rejectWithResponse = this.get('rejectWithResponse');
+      deprecateNotRejectingWithResponse(rejectWithResponse);
 
       const scopesString = makeArray(scope).join(' ');
       if (!isEmpty(scopesString)) {
         data.scope = scopesString;
       }
       this.makeRequest(serverTokenEndpoint, data, headers).then((response) => {
-        run(() => {
-          if (!this._validate(response)) {
-            reject('access_token is missing in server response');
-          }
+        deprecatePlainReturnValuesFromMakeRequest(response);
 
-          const expiresAt = this._absolutizeExpirationTime(response['expires_in']);
-          this._scheduleAccessTokenRefresh(response['expires_in'], expiresAt, response['refresh_token']);
-          if (!isEmpty(expiresAt)) {
-            response = assign(response, { 'expires_at': expiresAt });
-          }
+        this._parseResponse(response, rejectWithResponse).then((json) => {
+          run(() => {
+            if (!this._validateResponse(json)) {
+              reject('access_token is missing in server response');
+            }
 
-          resolve(response);
-        });
+            const expiresAt = this._absolutizeExpirationTime(json['expires_in']);
+            this._scheduleAccessTokenRefresh(json['expires_in'], expiresAt, json['refresh_token']);
+            if (!isEmpty(expiresAt)) {
+              json = assign(json, { 'expires_at': expiresAt });
+            }
+
+            resolve(json);
+          }).catch(reject);
+        }).catch(reject);
       }, (response) => {
-        run(null, reject, useResponse ? response : (response.responseJSON || response.responseText));
+        run(null, reject, rejectWithResponse ? response : (response.responseJSON || response.responseText));
```

Comment:
I think this code would now be accessible if the `fetch` in https://github.com/simplabs/ember-simple-auth/pull/2017/files#diff-931dc51edd167948e773858f72a47233R400 fails, and in that case we don't have the `responseJSON` or `responseText`. This is now done by `_parseResponse`

Reply:
I fixed that. That also required fixing the tests and adding previously missing responses to the failing tests (e.g. `return [201, { 'Content-Type': 'application/json' }, '{ "access_token": "secret token 2!" }'];`). These missing responses were previously causing an error already but because of the `response.responseJSON || response.responseText` which resulted to `undefined` on that error object, the tests didn't break 🤯